lib/player_ai_minimax.rb

```diff
@@ -0,0 +1,60 @@
+class PlayerAiMinimax
+  attr_reader :mark
+
+  def initialize(mark, rules)
+    @mark = mark
+    @rules = rules
+    @corners = [0,2,6,8]
+  end
+
+  def play_move(game_state)
+    board = game_state.game_board
+    return 0 if board.empty?
+    return 4 if opponent_played_corner_for_first_move(board)
+    return corner_move(board) if board.available_moves.size == 8
+    @current_player = game_state.player
+    @non_current_player = game_state.opponent
+    minimax(board, game_state.player, game_state.opponent)
+    @choice
+  end
+
+  private
+
+  def corner_move(board)
+    available_corners = @corners & board.available_moves
+    i = Random.rand(available_corners.size)
+    available_corners[i]
```

Comment:
Nitpick: you could consolidate these two lines by using [#sample](http://ruby-doc.org/core-2.2.0/Array.html#method-i-sample)